roles/openshift_hosted/tasks/registry/storage/s3.yml

```diff
@@ -21,13 +21,27 @@
       openshift_hosted_registry_storage_s3_cloudfront_keypairid and
       openshift_hosted_registry_storage_s3_cloudfront_privatekeyfile are required
 
-# Copy the cloudfront.pem to the host if the baseurl is given
-- name: Copy cloudfront.pem to the registry
-  copy:
-    src: "{{ openshift_hosted_registry_storage_s3_cloudfront_privatekeyfile }}"
-    dest: /etc/s3-cloudfront/cloudfront.pem
-    backup: true
-    owner: root
-    group: root
-    mode: 0600
+
+# Inject the cloudfront private key as a secret when required
+- block:
+
+    - name: Create registry secret for cloudfront
+      oc_secret:
+        state: present
+        namespace: "{{ openshift.hosted.registry.namespace | default('default') }}"
+        name: docker-registry-s3-cloudfront
+        contents:
+          path: cloudfront.pem
+          data: "{{ lookup('file', openshift_hosted_registry_storage_s3_cloudfront_privatekeyfile) }}"
+
+    - name: Add cloudfront secret to the registry deployment config
+      command: >
+        oc volume dc/docker-registry --add --name=cloudfront-vol
+        --namespace="{{ openshift.hosted.registry.namespace | default('default') }}"
+        -m /etc/origin --type=secret --secret-name=docker-registry-s3-cloudfront
+      register: cloudfront_vol_attach
```

Comment:
If I read the above code correctly, it will create a secret named "docker-registry-s3-cloudfront" with data "cloudfront.pem: xxxxxx", then attach it to registry. In registry container, the pem file will be located at "/etc/origin/cloudfront.pem". But according to https://github.com/openshift/openshift-ansible/blob/master/roles/openshift_hosted/templates/registry_config.j2#L81, in registry config file, it is trying to read key file from openshift_hosted_registry_storage_s3_cloudfront_privatekeyfile, it will fail to read that once openshift_hosted_registry_storage_s3_cloudfront_privatekeyfile is not "/etc/origin/cloudfront.pem". If keep the code in this PR, need modify https://github.com/openshift/openshift-ansible/blob/master/roles/openshift_hosted/templates/registry_config.j2#L81 to "privatekey: /etc/origin/cloudfront.pem". Am I right?